numba/ir_utils.py

```diff
@@ -1822,6 +1822,14 @@ def raise_on_unsupported_feature(func_ir, typemap):
     unsupported features.
     """
     gdb_calls = [] # accumulate calls to gdb/gdb_init
+    
+    for arg_name in func_ir.arg_names:
+        if arg_name in typemap and \
+           isinstance(typemap[arg_name], types.containers.UniTuple) and \
+           (typemap[arg_name].count > 1000):
+            raise TypingError(
+                    "Tuple '{}' length must be smaller "
+                    "than 1000".format(arg_name))
```

Comment:
I think that here:
1. It would be best to raise a `numba.errors.UnsupportedError`, the type isn't really the problem, it's that we're electing to not support doing this as it is problematic.
2. That the error class constructor, along with the message, should have the `loc=` kwarg set as `loc=func_ir.loc`, this will point to the location in the code where the problem is to help the user.
3. That the error message should be enhanced with a second section to explain why and offer a suggestion as an alternative, some suggestions for inclusion:
   * Generates prohibitively large LLVM IR which causes excess memory pressure and large compile times.
   * Alternative would be to use a list which does not suffer from this problem.

The general idea with the Numba frontend is to try and be as helpful as possible to the user, verbose error messages, hints and suggestions are all welcomed additions to these messages.